### PR TITLE
Migrate to Vite and Setup deployment to AWS

### PR DESCRIPTION
**Pull request by klaus-224**
Migration to Vite:
- created `vite.config.ts`
- added script in `index.html` to override behaviour of `SystemJS` enabling the use of `System.import()` for modules with the `System.register` format and `import()` for Native ES modules (which are used by Vite in dev) check this [MR](https://github.com/systemjs/systemjs/pull/2187/files#diff-106060984c76b31e91aabd02eae7200c0c25b3a81fb1564cd971836ae15a0f61)
- using `SystemJS` because it supports external import maps
- supports HMR
- using Vitest for unit-tests
- NOTE: Linting does not work right now because Babel was removed -> TODO add typescript parser

Deployment:
- infrastructure includes CloudFront and S3 -> no domain right now

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,93 +1,129 @@
+version: 2.1
+executors:
+  default:
+    docker:
+      - image: cimg/node:17.3
```
Comment (alex-hladun) [inline .circleci/config.yml]: should use node 18 if possible. The even numbers of node will have LTS

https://nodesource.com/blog/understanding-how-node-js-release-lines-work/

File: import-maps/production.import-map.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "imports": {
+    "single-spa": "https://cdn.jsdelivr.net/npm/single-spa@5.9.0/lib/system/single-spa.min.js",
+    "react": "https://cdn.jsdelivr.net/npm/react@17.0.2/umd/react.development.min.js",
```
Comment (alex-hladun) [inline import-maps/production.import-map.json]: Don't want to use development React / react-dom builds in prod

File: infrastructure/lib/configuration.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    account: process.env.CDK_DEFAULT_ACCOUNT,
+    region: process.env.CDK_DEFAULT_REGION,
+  },
+};
+
+export const configuration: { [key: string]: Configuration } = {
+  ["production"]: {
+    ...commonConfig,
+    stageName: "production",
+    certificateArn:
```
Comment (alex-hladun) [inline infrastructure/lib/configuration.ts]: we should avoid hard-coding this if possible...

File: src/tests/example.test.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+function addNumbers(a, b) {
+  return a + b;
+}
+
+describe("math", () => {
+  test("adds 1 + 2 to equal 3", () => {
+    const result = addNumbers(1, 2);
+
+    expect(result).toBe(3);
```
Comment (alex-hladun) [inline src/tests/example.test.ts:9]: verified this myself, checks out

**Review by alex-hladun — CHANGES_REQUESTED** [review]
couple comments

File: src/tests/example.test.ts (hunk 4 above)
Comment (klaus-224) [reply src/tests/example.test.ts:9]: Cool, I was about to bust out the TI-89 to verify myself

**Review by alex-hladun — APPROVED** [review]

**Pull request merged**